### ISSUE TITLE
Multi Query on Async Fileds

### ISSUE THREAD

**Issue by manoochehrkateb** [graphql-dotnet/graphql-dotnet#1870]
Let say I have 2 Async fields like :
 ```
FieldAsync<x>(....)
FieldAsync<y>(....)
```
and when I want to consume those like 
```
query{
  x : getListCustomer()
  y : customerReport_Sms()
}
```
its throw an exception 

> Entity Framework Core: A second operation started on this context before a previous operation completed

**But when I changed them to non-async filed everything works fine .
My dbContext is Transient**

**Comment by Shane32** [comment]
Is your dbContext created via the constructor of the graphtype?  If so, since the graphtype is (probably) a singleton, the dbcontext will be the same instance for both fields.  You will need to create a separate dbcontext for each field.  For example, you could write `var db = context.RequestServices.GetRequiredService<MyDbContext>()` to get a local copy of the db to use within each field resolver.  Then, with a transient dbcontext, you can run queries in parallel.

**Comment by Shane32** [comment]
When they are not async, the fields will not run in parallel, even with a ParallelExecutionStrategy.

**Comment by manoochehrkateb** [comment]
Thanks @Shane32 . Seems like I have shared dbContext . But my graphtype is Transient . 

> When they are not async, the fields will not run in parallel, even with a ParallelExecutionStrategy

Can I change this ParallelExecutionStrategy ? 

.

**Comment by sungam3r** [comment]
Duplicate of #1310

**Comment by sungam3r** [comment]
Yes, see https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Execution/DocumentExecuter.cs#L277

**Comment by Shane32** [comment]
> Thanks @Shane32 . Seems like I have shared dbContext . But my graphtype is Transient .
> 
> > When they are not async, the fields will not run in parallel, even with a ParallelExecutionStrategy
> 
> Can I change this ParallelExecutionStrategy ?
> 
> .

You could wrap the code in Task.Run but you would then run into the same problems you are having with the Async methods. I’m just pointing out why you did not notice the problem when using non-async methods. 

You can use a SerialExecutionStrategy if you want to force both async and sync calls to run in series, eliminating your immediate problem.
